tests/suites/os/tests/secureboot/index.js

```diff
@@ -184,6 +173,113 @@ class uefiSecureBoot extends secureBoot {
+			this.link
+		)
+		const key = reg
+			.replace(/0x/g, '') // remove 0x prefix
+			.trim()							// remove extra spaces
+			.split(/\s+/)				// split by whitespace
+			.slice(7, 15)				// take elements 8 to 15
+			.join('');          // concatenate
+		return key;
+	}
+
+	async readOTPReg(reg) {
+		let out = await this.worker.executeCommandInHostOS(
+			['vcgencmd', 'otp_dump', '2>/dev/null'],
+			this.link
+		)
+		for ( const line of out.split('\n')) {
+			if (line.startsWith(reg)) {
+				return line.split(':')[1].trim();
+			}
+		}
+	}
+
+	async checkRSAkey() {
+		for (let reg = 47; reg <= 54; reg++) {
+			let out = await this.readOTPReg(reg)
+			if ( out && out != "00000000") {
+				return true
+			}
+		}
+		return false
+	}
+
+	async isSecureBootEnabled() {
+		// OTP is programmed with customer key digest and private key
+		if ( await this.checkRSAkey() ) {
+			const key = await this.readPrivateKey();
+			if ( key.replace(/0/g, '').trim().length > 0 ) {
+				return true;
+			}
+		}
+	}
+
+	async waitForFailedBoot(){
+		// confirm that eth link is high
+		// we don't know what the ethernet interface connecting worker to DUT is, so first find that, using the autokit WIRED_IFACE var
+		let iface = await this.worker.executeCommandInWorker(`env | grep WIRED | awk -F'=' '{print $2}'`);
+
+		console.log(`Watching interface: ${iface}`)
+		// confirm that the link is high - confirming that the board is on
+		await this.utils.waitUntil(async() => {
+			console.log('Confirming eth link is present...')
+			let ethLink = await this.worker.executeCommandInWorker(`cat /sys/class/net/${iface}/carrier`);
+			console.log(`Eth link is: ${ethLink}`)
+			return ethLink === '1';
+		})
+
+		// SSH access shouldn't be possible - as the device shouldn't be booting?
+		// the CM4 takes a while to get into a state where we can SSH into it anyway after powering on - so we want to try a good number of times.
+		try {
+			await this.worker.executeCommandInHostOS('echo -n pass',this.link, { max_tries: 20, interval: 1000 });
+			// if we manage to SSH into the DUT, then something has gone wrong - so throw an error which will fail the test
+			throw new Error(`DUT was still reachable over SSH`)
+		} catch (e){
+			// we want / are expecting a failure here, so if the SSH fails after the specified number of retries, we want to return
+			console.log(`DUT was not reachable over SSH`)
+			return
+		}
+	}
+
+	async testBootloaderIntegrity() {
+		/* Modify boot.sig to fail authentication */
+		await this.worker.executeCommandInHostOS(
+			"sed -i -e '1s/^.\\(.*\\)$/\\1/' /mnt/rpi/boot.sig",
+			this.link
+		)
+		await this.worker.executeCommandInHostOS('reboot', this.link)
+	    await this.test.resolves(
+			this.waitForFailedBoot(),
+			'Bootloader will not load configuration that fails signature verification',
+		)
+		await this.resetDUT();
+	}
+
+	async testBootloaderConfigIntegrity() {
+		/* Modify config.txt inside boot.img */
```

Comment:
In the test suite, I've just noticed that for all devices with a slug that starts with `raspberry*` we've writing to the `config.txt` of the OS image before flashing it to the DUT: https://github.com/balena-os/meta-balena/blob/master/tests/suites/os/suite.js#L54 + https://github.com/balena-os/meta-balena/blob/master/tests/suites/os/suite.js#L24

Will this cause a problem with a locked device?

Reply:
hey @rcooke-warwick this will be the `config.txt` in the flasher image so it won't be a problem. Depending on how the bootloader is configured, it will either use it, or ignore it if it is configured to boot using `boot.img`.